packages/salesforcedx-vscode-apex-debugger/test/unit/adapter/apexDebugForTest.ts

```diff
@@ -0,0 +1,86 @@
+/*
+ * Copyright (c) 2017, salesforce.com, inc.
+ * All rights reserved.
+ * Licensed under the BSD 3-Clause license.
+ * For full license text, see LICENSE.txt file in the repo root or https://opensource.org/licenses/BSD-3-Clause
+ */
+
+import { DebugProtocol } from 'vscode-debugprotocol';
+import {
+  ApexDebug,
+  LaunchRequestArguments
+} from '../../../src/adapter/apexDebug';
+import { SessionService } from '../../../src/core/sessionService';
+import { CommandOutput } from '../../../src/utils/commandOutput';
+
+export class ApexDebugForTest extends ApexDebug {
+  private receivedResponse: DebugProtocol.Response;
+  private receivedEvents: DebugProtocol.Event[];
+
+  constructor(sessionService: SessionService, timeout?: number) {
+    super();
+    this.receivedEvents = new Array();
+    this.mySessionService = sessionService;
+  }
+
+  public getResponse(): DebugProtocol.Response {
+    return this.receivedResponse;
+  }
+
+  public getEvents(): DebugProtocol.Event[] {
+    return this.receivedEvents;
+  }
+
+  public sendResponse(response: DebugProtocol.Response): void {
+    if (this.receivedResponse) {
+      throw new Error('Should not receive more than one response');
+    }
+    this.receivedResponse = response;
+  }
+
+  public sendEvent(event: DebugProtocol.Event): void {
+    this.receivedEvents.push(event);
+  }
+
+  public initializeReq(
```

Comment:
I'm confused by these methods. They all seem to delegate back up to super.
What's the purpose of this class then? Is it so that we have a method to grab a handle to for mocking purposes?